src/lib.rs

```diff
@@ -258,7 +260,19 @@ decl_error! {
         RemoteAttestationVerificationFailed,
         RemoteAttestationTooOld,
         ///The enclave cannot attest, because its building mode is not allowed
-        SgxModeNotAllowed
+        SgxModeNotAllowed,
+        ///The enclave is not registered
+        EnclaveIsNotRegistered,
+        ///The bonding account doesn't match the enclave
+        BondingAccountNotMatchEnclave,
+        ///The shard doesn't match the enclave
+        ShardNotMatchEnclave,
+        ///The worker url is too long
+        EnclaveUrlTooLong,
+        ///The RA report is too long
+        RaReportTooLong,
+        ///The enclave doesn't exists
```

Comment:
Short naming hints: In general, it is advised to use `to be` phrases if possible as opposed to `to do` phrases and leave out the verb. So I have some preferences to `NotMatch`, either:

* `NotEqualTo`
* `NotMatching`

Reply:
As only the default shard equals the MRENCLAVE, I prefer NotMatching. See: https://github.com/integritee-network/pallet-teerex/issues/7#issuecomment-937723968
So, what about ShardNotMatchingEnclave and BondingAccountNotMatchingEnclave ?